assignments/lambda-classes.js

```diff
@@ -1 +1,191 @@
 // CODE here for your Lambda Classes
+
+/**
+ *  This function takes two arguments,
+ *  a maximum and a minimum number and returns a random number
+ *  between the given range. The returned value is inclusive of
+ *  the max and min numbers.
+ */
+const randomNumberGenerator = (min, max) => {
+  return Math.floor(Math.random() * (max - min + min) + min);
+};
+
+class Person {
+  constructor(name, age, location, gender) {
+    this.name = name;
+    this.age = age;
+    this.location = location;
+    this.gender = gender;
+  }
+
+  speak() {
+    return `Hello my name is ${this.name}, I am from ${this.location}`;
+  }
+}
+
+class Instructor extends Person {
+  constructor({
+    name,
+    age,
+    location,
+    gender,
+    specialty,
+    favLanguage,
+    catchPhrase
+  }) {
+    super(name, age, location, gender);
+    this.specialty = specialty;
+    this.favLanguage = favLanguage;
+    this.catchPhrase = catchPhrase;
+  }
+
+  demo(subject) {
+    return `Today we are learning about ${subject}`;
+  }
+
+  grade(student, subject) {
+    return `${student.name} receives a perfect score on ${subject}`;
+  }
+
+  gradeStudent(student) {
+    const oldGrade = student.grade;
+    student.grade = randomNumberGenerator(1, 100);
+
+    return `${student.name} now has new grade of ${
+      student.grade
+    }, old grade was ${oldGrade}`;
+  }
+}
+
+class Student extends Person {
+  constructor(studentAttr) {
+    super(
+      studentAttr.name,
+      studentAttr.age,
+      studentAttr.location,
+      studentAttr.gender
+    );
+    this.previousBackground = studentAttr.previousBackground;
+    this.className = studentAttr.className;
+    this.favSubjects = studentAttr.favSubjects;
+    this.grade = studentAttr.grade;
+    this.hasGraduated = studentAttr.hasGraduated;
+  }
+
+  listsSubjects() {
+    this.favSubjects.map(subject => console.log(subject));
```

Comment:
You get undefined when you log the function to the console.
You could want to refactor this function to return a string with all the values of your array.

Reply:
Thanks.
I have refactored it to return a string instead.